### DISCUSSION TITLE
Updating the objects after renaming them

### DISCUSSION THREAD

**Discussion by SK-mechcoder** [ansys/pyaedt#5486] [General]
Hello, Good day,

This is the code I developed, the idea here is that I need to rename the object after importing it into the Ansys electronics desktop, but 
Renaming through ChangeProperty works to update the **name of the object**, but it does not update the object’s reference in the **model_objects** list.

how to achieve that the object's reference is also get updated.

I attached my code below for reference.

```
from ansys.aedt.core import Maxwell3d

SR6_relay = Maxwell3d(project="SR6_relay", non_graphical=False, new_desktop=True)
Iteration_1=SR6_relay.rename_design("Iteration_1")
relay=SR6_relay.import_3d_cad("SR6.stp",create_group=False)
imported_bodies=SR6_relay.modeler.model_objects
##
##
#Dynamic change of object for coil
print(imported_bodies)

for i in imported_bodies:
    if 'coil' in i.lower() or 'spule' in i.lower():
        obj=SR6_relay.modeler.get_object_from_name(assignment=i)
        print(obj)
        SR6_relay.change_property(aedt_object=SR6_relay.oeditor,tab_name="Geometry3DAttributeTab",property_object=i,property_name="Name",property_value="coil")
        print(imported_bodies)
    else:
        print('there is no change')
SR6_relay.release_desktop(close_projects=False, close_desktop=False)
```

Many thanks for considering my request.

**Comment by SK-mechcoder** [comment]
this is the message i'm getting, 

even though the property name is changed, when I'm printing imported bodies I'm getting this list 
**['15587072_SR6II_ARMATURE_CALIB', '15587072_SR6II_ARMATURE_CALIB_1', '15587062_SR6II_CORE_CALIB_1_1', '15587052_SR6II_YOKE_CALIB_1_1', '15587161_SR6II_COIL_1_1']**
instead of 
**['15587072_SR6II_ARMATURE_CALIB', '15587072_SR6II_ARMATURE_CALIB_1', '15587062_SR6II_CORE_CALIB_1_1', '15587052_SR6II_YOKE_CALIB_1_1', 'coil']**

**C:\Users\TE544300\python_venv_automation\Scripts\python.exe C:\To_be_copied\automation\aedt\ProgramsforMaxwell\trialdelete.py 
PyAEDT INFO: Python version 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
PyAEDT INFO: PyAEDT version 0.11.5.
PyAEDT INFO: Initializing new Desktop session.
PyAEDT INFO: Log on console is enabled.
PyAEDT INFO: Log on file C:\Users\TE544300\AppData\Local\Temp\pyaedt_TE544300_c363ef37-01ac-43e5-93e1-412065ca67f9.log is enabled.
PyAEDT INFO: Log on AEDT is enabled.
PyAEDT INFO: Debug logger is disabled. PyAEDT methods will not be logged.
PyAEDT INFO: Launching PyAEDT with gRPC plugin.
PyAEDT INFO: New AEDT session is starting on gRPC port 54339
PyAEDT INFO: AEDT installation Path C:\ENGapps\AnsysEM\v232\Win64
PyAEDT INFO: Ansoft.ElectronicsDesktop.2023.2 version started with process ID 30184.
PyAEDT INFO: Project SR6_relay has been created.
PyAEDT INFO: No design is present. Inserting a new design.
PyAEDT INFO: Added design 'Maxwell 3D_10D' of type Maxwell 3D.
PyAEDT INFO: Aedt Objects correctly read
PyAEDT INFO: Modeler class has been initialized! Elapsed time: 0m 1sec
PyAEDT INFO: Step file SR6.stp imported
['15587072_SR6II_ARMATURE_CALIB', '15587072_SR6II_ARMATURE_CALIB_1', '15587062_SR6II_CORE_CALIB_1_1', '15587052_SR6II_YOKE_CALIB_1_1', '15587161_SR6II_COIL_1_1']
there is no change
there is no change
there is no change
there is no change
15587161_SR6II_COIL_1_1
PyAEDT INFO: Property Name changed correctly.
['15587072_SR6II_ARMATURE_CALIB', '15587072_SR6II_ARMATURE_CALIB_1', '15587062_SR6II_CORE_CALIB_1_1', '15587052_SR6II_YOKE_CALIB_1_1', '15587161_SR6II_COIL_1_1']
PyAEDT INFO: Desktop has been released.

Process finished with exit code 0**

**Comment by Samuelopez-ansys** [comment]
Hi @SK-mechcoder,

You are used to the AEDT API, which indeed is not object-oriented. So when you modify a property of an object in the way you did, PyAEDT is not able to grab the update, and the only solution would be save the project and reload the PyAEDT object again, which I do not like because it is time consuming.

I would use a more pythonic way, which is one of the main "powers" of PyAEDT:

```
for i in imported_bodies:
    if 'wg' in i.lower():
        obj = SR6_relay.modeler.get_object_from_name(assignment=i)
        obj.name = "coil"
    else:
        print('there is no change')

SR6_relay.release_desktop(close_projects=False, close_desktop=False)
app.release_desktop(False, False)
```

As you can see "obj" is a python object which has all the information, properties and methods to manipulate it.

Thanks,
Samuel

**Comment by SK-mechcoder** [comment]
Hello Samuelopez,

I tried this code, the obj name has changed in the Electronics desktop but the imported bodies still produce same out before and after the property change. I attached my code below.

```
from ansys.aedt.core import Maxwell3d

SR6_relay = Maxwell3d(project="SR6_relay", non_graphical=False, new_desktop=True)
Iteration_1=SR6_relay.rename_design("Iteration_1")
relay=SR6_relay.import_3d_cad("SR6.stp",create_group=False)
imported_bodies=SR6_relay.modeler.model_objects
##
##
#Dynamic change of object for coil
print(imported_bodies)

for i in imported_bodies:
    if 'coil' in i.lower() or 'spule' in i.lower():
        obj=SR6_relay.modeler.get_object_from_name(assignment=i)
        obj.name="coil"
    else:
        print('there is no change')
print(imported_bodies)
SR6_relay.release_desktop(close_projects=False, close_desktop=False)
```

here are the messages from the console.
> 
> C:\Users\TE544300\python_venv_automation\Scripts\python.exe C:\To_be_copied\automation\aedt\ProgramsforMaxwell\trialdelete.py 
> PyAEDT INFO: Python version 3.10.9 (tags/v3.10.9:1dd9be6, Dec  6 2022, 20:01:21) [MSC v.1934 64 bit (AMD64)]
> PyAEDT INFO: PyAEDT version 0.11.5.
> PyAEDT INFO: Initializing new Desktop session.
> PyAEDT INFO: Log on console is enabled.
> PyAEDT INFO: Log on file C:\Users\TE544300\AppData\Local\Temp\pyaedt_TE544300_974ac8f5-da42-4d01-866d-81bde64369cd.log is enabled.
> PyAEDT INFO: Log on AEDT is enabled.
> PyAEDT INFO: Debug logger is disabled. PyAEDT methods will not be logged.
> PyAEDT INFO: Launching PyAEDT with gRPC plugin.
> PyAEDT INFO: New AEDT session is starting on gRPC port 53213
> PyAEDT INFO: AEDT installation Path C:\ENGapps\AnsysEM\v232\Win64
> PyAEDT INFO: Ansoft.ElectronicsDesktop.2023.2 version started with process ID 29408.
> PyAEDT INFO: Project SR6_relay has been created.
> PyAEDT INFO: No design is present. Inserting a new design.
> PyAEDT INFO: Added design 'Maxwell 3D_VYB' of type Maxwell 3D.
> PyAEDT INFO: Aedt Objects correctly read
> PyAEDT INFO: Modeler class has been initialized! Elapsed time: 0m 0sec
> PyAEDT INFO: Step file SR6.stp imported
> ['15587072_SR6II_ARMATURE_CALIB', '15587072_SR6II_ARMATURE_CALIB_1', '15587062_SR6II_CORE_CALIB_1_1', '15587052_SR6II_YOKE_CALIB_1_1', '15587161_SR6II_COIL_1_1']
> there is no change
> there is no change
> there is no change
> there is no change
> ['15587072_SR6II_ARMATURE_CALIB', '15587072_SR6II_ARMATURE_CALIB_1', '15587062_SR6II_CORE_CALIB_1_1', '15587052_SR6II_YOKE_CALIB_1_1', '15587161_SR6II_COIL_1_1']
> PyAEDT INFO: Desktop has been released.
>Process finished with exit code 0

and here is the screenshot
![image](https://github.com/user-attachments/assets/92d0ded5-775f-428c-9b3f-effeb37461f6)

as you can see the references are not updated.

**Comment by Samuelopez-ansys** [comment]
@SK-mechcoder You need to get the list of objects again:

new_imported_bodies = SR6_relay.modeler.model_objects
print(new_imported_bodies )

**Comment by SK-mechcoder** [comment]
Hello @Samuelopez-ansys ,

it worked perfectly, thanks for the help.
